### PR TITLE
Add support for pydevd and pydevd-pycharm

### PR DESCRIPTION
**Pull request by briandealwis**
Relevant: https://github.com/GoogleContainerTools/skaffold/issues/2202, https://github.com/GoogleContainerTools/skaffold/issues/4652, https://github.com/GoogleContainerTools/skaffold/issues/4088

This PR adds `pydevd` and `pydevd` to the debug support images.  It also introduces a new python launcher in `<root>/python/launcher` that determines the python executable based on the original command line, and configures the PYTHONPATH to point to the suitable pydevd/debugpy/ptvsd installation.
```
launcher --mode <pydevd|debugpy|ptvsd> --port p [--wait] -- original-command-line ...
```

The `--wait` flag is currently not functional for pydevd.

### CODE REVIEW

File: python/helper-image/Dockerfile [hunk 1]
```diff
@@ -1,14 +1,54 @@
...
 RUN PYTHONUSERBASE=/dbgpy pip install --user ptvsd debugpy
+RUN PYTHONUSERBASE=/dbgpy/pydevd/python3.8 pip install --user pydevd
+RUN PYTHONUSERBASE=/dbgpy/pydevd-pycharm/python3.8 pip install --user pydevd-pycharm
 
 FROM python:3.9 as python39
 RUN PYTHONUSERBASE=/dbgpy pip install --user ptvsd debugpy
+RUN PYTHONUSERBASE=/dbgpy/pydevd/python3.9 pip install --user pydevd
+RUN PYTHONUSERBASE=/dbgpy/pydevd-pycharm/python3.9 pip install --user pydevd-pycharm
+
+FROM golang:1.14.1 as build
```
Comment (briandealwis) [inline python/helper-image/Dockerfile:48]: I'll switch this to `golang:1.16` in a subsequent commit — there are a few other uses in this repo.

**Comment by aaron-prindle** [comment]
Might be helpful to add a blurb or link the changes required to update `transform_python.go`:
https://github.com/GoogleContainerTools/skaffold/blob/master/pkg/skaffold/debug/transform_python.go#L58:6

Part of fix for:
https://github.com/GoogleContainerTools/skaffold/issues/2202

File: python/helper-image/launcher/cmd.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+var newConsoleCommand = createConsoleCommand
+
+// commander is a subset of exec.Cmd
+type commander interface {
+	Run() error
+	Output() ([]byte, error)
+	CombinedOutput() ([]byte, error)
+}
+
+var _ commander = (*exec.Cmd)(nil)
```
Comment (aaron-prindle) [inline python/helper-image/launcher/cmd.go:39]: nit: might be helpful to add a comment that this is check verifying that exec.Cmd satisfies the `commander` interface

File: python/helper-image/launcher/cmd_test.go [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+type fakeCmd struct {
+	mode     string
+	cmdline  []string
+	exitCode int
+	output   string
+}
+
+var _ commander = (*fakeCmd)(nil)
```
Comment (aaron-prindle) [inline python/helper-image/launcher/cmd_test.go:50]: nit: might be helpful to add a comment that this is check verifying that fakeCmd satisfies the `commander` interface

**Comment by aaron-prindle** [comment]
LGTM

**Review by aaron-prindle — APPROVED** [review]

File: python/helper-image/launcher/launcher_test.go [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+				AndRunCmd([]string{"python", "-m", "ptvsd", "--host", "localhost", "--port", "2345", "--wait", "app.py"}),
+			expected: pythonContext{debugMode: "ptvsd", port: 2345, wait: true, args: []string{"python", "-m", "ptvsd", "--host", "localhost", "--port", "2345", "--wait", "app.py"}, env: env{"PYTHONPATH": dbgRoot + "/python/lib/python3.7/site-packages"}},
+		},
+		{
+			description: "pydevd",
+			pc:          pythonContext{debugMode: "pydevd", port: 2345, wait: false, args: []string{"python", "app.py"}, env: nil},
+			commands: RunCmdOut([]string{"python", "-V"}, "Python 3.7.4\n").
+				AndRunCmd([]string{"python", "-m", "pydevd", "--port", "2345", "--server", "--file", "app.py"}),
+			expected: pythonContext{debugMode: "pydevd", port: 2345, wait: false, args: []string{"python", "-m", "pydevd", "--port", "2345", "--server", "--file", "app.py"}, env: env{"PYTHONPATH": dbgRoot + "/python/pydevd/python3.7/lib/python3.7/site-packages"}},
+		},
```
Comment (aaron-prindle) [inline python/helper-image/launcher/launcher_test.go:277]: Might be helpful to have the full set of modes in the test cases here:
debugpy, ptvsd, pydevd, pydevd-pycharm

Comment (briandealwis) [reply python/helper-image/launcher/launcher_test.go:277]: The pydevd-pycharm may be dropped — we haven't been able to make it work for PyCharm.

**Pull request merged**